test/UnitTest/Extensions/HttpClientExtensionTest.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Extensions;
+
+using Moq;
+using Moq.Protected;
+
+using Xunit;
+
+namespace Altinn.Platform.Storage.Tests.Extensions
+{
+    public class HttpClientExtensionTest
+    {
+        [Fact]
+        public async Task PostAsync_ShouldAddAuthorizationHeaderAndReturnHttpResponseMessage()
+        {
+            // Arrange
+            HttpResponseMessage httpResponseMessage = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent("dummyData", Encoding.UTF8, "application/json")
+            };
+
+            Mock<HttpMessageHandler> handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
```

Comment:
Instead of mocking a HttpMessageHandler like this, we've started implementing a [DelegatingHandlerStub](https://github.com/Altinn/altinn-notifications/blob/main/test/Altinn.Notifications.Tests/DelegatingHandlerStub.cs) class. Take a look at how we're using it in the notifications solution.